src/store/mod.rs

```diff
@@ -1,11 +1,13 @@
 use crate::model::Pod;
-use std::{collections::BTreeMap, error::Error};
+use anyhow::Result;
+use std::collections::BTreeMap;
 
 pub trait Store {
-    fn save_pod(&self, pod: &Pod) -> Result<(), Box<dyn Error>>;
-    fn load_pod(&self, name: &str, version: &str) -> Result<Pod, Box<dyn Error>>;
-    fn list_pod(&self) -> Result<BTreeMap<String, Vec<String>>, Box<dyn Error>>;
-    fn delete_pod(&self, name: &str, version: &str) -> Result<(), Box<dyn Error>>;
+    fn save_pod(&mut self, pod: &Pod) -> Result<()>;
+    fn load_pod(&mut self, name: &str, version: &str) -> Result<Pod>;
```

Comment:
This PR seems to also address making `Annotation` optional. If so, I would expect `load_pod` to handle loading with name/version (with annotation) or using hash (w/o annotation). The corresponding [issue](https://github.com/walkerlab/orcapod/issues/27) has more info on these cases.

My recommendation would actually be to move that feature into a separate PR since it is starting to become too much for one PR. That way it isn't all-or-nothing.

Reply:
A few of the changes deals provides the ability to do that a bit easier in the future, but since we already decided on going with enum to identify it, I can code it in by tonight.

Reply:
Update pull request to have handle this case along with tests.